Kernel/Memory/MemoryManager.cpp

```diff
@@ -744,22 +744,22 @@ ErrorOr<NonnullOwnPtr<Region>> MemoryManager::allocate_contiguous_kernel_region(
     return allocate_kernel_region_with_vmobject(range, move(vmobject), name, access, cacheable);
 }
 
-ErrorOr<NonnullOwnPtr<Memory::Region>> MemoryManager::allocate_dma_buffer_page(StringView name, Memory::Region::Access access, RefPtr<Memory::PhysicalPage>& dma_buffer_page)
+ErrorOr<NonnullOwnPtr<Memory::Region>> MemoryManager::allocate_dma_buffer_page(StringView name, Memory::Region::Access access, RefPtr<Memory::PhysicalPage>& dma_buffer_page, Region::Cacheable cacheable)
```

Comment:
I think having a DMA region cacheable is pointless in terms of how Serenity deals with paging, as well as the usage in this context (a DMA buffer). Is there a point of marking these `cacheable`? I'm pretty sure that this form of page level caching will probably be ignored by the processor, as the DMA controller is actually doing a physical memory transfer, meaning anything in the cache will probably immediately go stale. There's also the fact that the default value passed into `Region::Region` is already `Cacheable::Yes`, so this is largely superfluous.

In terms of Serenity's Virtual Memory system, we by default have PAE enabled for the i686 version of the Kernel. According the Intel Reference Manual 3A, 2-16

```
This bit is not used if paging is disabled, with PAE paging, or with IA-32e paging if CR4.PCIDE=1.
```
The processor will ignore this bit entirely. https://www.intel.com.au/content/www/au/en/architecture-and-technology/64-ia-32-architectures-software-developer-vol-3a-part-1-manual.html

Reply:
Thanks a lot for the comment! I had a vague feeling about what you mentioned when I added this change but now it is clear to me now. 
I added the extra parameter to accommodate AHCIPort DMA allocation where it explicitly sets the cache to be disabled. 
We have two options now:
- Get rid of this parameter for dma buffer helper (as it was there before) because anyway this bit is ignored.
Or
- Add this parameter with default value to `Cacheable::No` so that we make this explicit as it is a DMA region. 

I feel it is better to be explicit instead of relying on a processor dependent behavior. What do you think?

Reply:
> I feel it is better to be explicit instead of relying on a processor dependent behavior. What do you think?

Agreed, I'm of the opinion that having an explicit argument is much nicer for something like this than not, hence I'd be more inclined to think option two as you've suggested above is best, though I'll leave to the resident C++ nerds, so calling Dr. @alimpfard

Reply:
I think it's okay to have the extra argument with a default of `Yes` (which is the old behaviour), and set it to `No` wherever the PCD bit is needed to be set.

Reply:
As we are always going to set this to `No`, we don't even need to have an extra parameter in the DMA helper API so that the user of the API doesn't even need to worry about it. 
Instead the call could be as follows:
```
 auto region_or_error = allocate_kernel_region(dma_buffer_page->paddr(), PAGE_SIZE, name, access, Region::Cacheable::No);
```
In that way, we don't even need the extra param in the API for DMA allocation.

Reply:
@alimpfard So you suggest all the callers of the API explicitly set the value to `Region::Cacheable::No` instead of the API doing it for them as that is the default behavior we want for a DMA region? 

> As we are always going to set this to `No`, we don't even need to have an extra parameter in the DMA helper API so that the user of the API doesn't even need to worry about it. Instead the call inside the function to `allocate_kernel_region` could be as follows:
> 
> ```
>  auto region_or_error = allocate_kernel_region(dma_buffer_page->paddr(), PAGE_SIZE, name, access, Region::Cacheable::No);
> ```
> 
> In that way, we don't even need the extra param in the API for DMA allocation.

What do you think about this suggestion?

Reply:
I'm only suggesting a default of `Yes` to keep the previous default behaviour, if that's not needed, then a default of `No` would work just as well.